### PR TITLE
feat(docs): Add Chinese Localization Update for Tact

### PR DESCRIPTION
**Pull request by sansx**
## Issue
This PR resolves the Chinese localization tasks for Tact, which related to issue #934. The entire translation has been completed using Crowdin's AI-powered translation tool.

Preview Link: 
https://tact-git-l10nmain-townsquarexyz.vercel.app/zh-cn/?_vercel_share=iQ0dtxO3eY4jES4Nl0IuLiK7ll6apcut

Notes:
The translation is **fully AI-generated** and will require proofreading to ensure accuracy and contextual appropriateness. 

We'll provide further proofreading in future updates. Additionally, if anyone is interested, you are welcome to [join our Crowdin project](https://crowdin.com/project/tact) to help with proofreading and finalizing the translations.

## Checklist

- [x] I have updated CHANGELOG.md
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Comment by novusnota** [comment]
Great, thanks for the PR! Regarding formatting issues in `crowdin.yml`, run the `yarn fmt:check -w` to resolve those.

Because of the PR from the org, Tact team cannot really edit your branch, so I'm going to provide suggestions in text form only:

1. Double-check that all links lead to proper places — running `yarn build` inside `docs/` folder will build the docs AND check that all the internal links point to places that exist. This is especially important since AI can hallucinate them on the fly — that's probably the reason why there was a new anchor link, that failed the CI.

2. Double-check that formatters on all pages match the following schema: https://starlight.astro.build/reference/frontmatter/. Locally running `yarn build` would find any issues.

3. Double-check that MDX components, such as [Steps](https://starlight.astro.build/components/steps/), [Tabs](https://starlight.astro.build/components/tabs/), [Asides](https://starlight.astro.build/guides/authoring-content/#asides) (with `:::` syntax), [Link Cards](https://starlight.astro.build/components/link-cards/) and the like continue to work the same. Locally running `yarn build` would find any issues.

Points 2 and 3 seem to be fine from my viewpoint, but double-checking is always a plus :)

Mind you, that there are LOTS of pages in your PR where English and Chinese sentences or paragraphs are mixed together or follow each other. I'm not sure that is a good thing, because if the whole page remained in English, one could've always just run Google Translate or other website translation tool and get decent results. But with mixed content, that's no longer an option — so it's probably better to exclude such pages until a proper translation is present there.

You don't have to provide translations for all pages - the documentation automatically generates fallback English pages with a note at the top about the missing translation. And to add to my point about website translation tools, I believe that when it comes to good localization, quality is much more important than quantity. So if you are able to provide only 10 complete and good translations, it's much better than 100 lackluster AI ones :)

**Comment by sansx** [comment]
@novusnota Hey, thanks so much for all these pointers! We've taken care of the issues you've mentioned, and we’re lining up resources to make further tweaks once this PR is merged.

As we said, our ultimate goal is to keep making the docs better and better, so we’ll keep iterating toward a version everyone finds clear and useful. Thanks again for your input—it’s always appreciated! 👍

**Comment by novusnota** [comment]
@sansx awesome! Let's fix the formatting by running `yarn fmt:check -w` (from the root of the repo) and I'll merge this PR, so that you can continue iterating on Chinese docs further :)

**Comment by sansx** [comment]
@novusnota thanks! I just finished this check, updated the fixed version in the last commit, and it gave me this as result now:
```
Checking formatting...
All matched files use Prettier code style!
✨  Done in 2.81s.
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Ability to specify a compile-time method ID expression for getters: PR [#922](https://github.com/tact-lang/tact/pull/922) and PR [#932](https://github.com/tact-lang/tact/pull/932)
 - Destructuring of structs and messages: PR [#856](https://github.com/tact-lang/tact/pull/856)
 - Docs: automatic links to Web IDE from all code blocks: PR [#994](https://github.com/tact-lang/tact/pull/994)
+- Docs: Added Chinese translation of the documentation: PR [#942](https://github.com/tact-lang/tact/pull/942)
```
Comment (novusnota) [inline CHANGELOG.md]:
```suggestion
- Docs: initial semi-automated Chinese translation of the documentation: PR [#942](https://github.com/tact-lang/tact/pull/942)
```

Comment (sansx) [reply CHANGELOG.md]: done :D

**Review by novusnota — APPROVED** [review]

**Pull request merged**